backend/internal/controller/adapter/shot_message.go

```diff
@@ -0,0 +1,74 @@
+package adapter
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/HackYourCareer/SmartKickers/internal/config"
+	"github.com/HackYourCareer/SmartKickers/internal/model"
+)
+
+type tableShotParams struct {
```

Comment:
```suggestion
type shotParams struct {
```
I would consider renaming this structure to just "shotParams". Shot will always come from the table anyway.